src/Avalonia.Vulkan/VulkanInstance.cs

```diff
@@ -0,0 +1,160 @@
+            }
+
+            var ppEnabledExtensions = stackalloc IntPtr[enabledExtensions.Count];
+            var ppEnabledLayers = stackalloc IntPtr[enabledLayers.Count];
+
+            for (var i = 0; i < enabledExtensions.Count; i++)
+                ppEnabledExtensions[i] = Marshal.StringToHGlobalAnsi(enabledExtensions[i]);
+
+            for (var i = 0; i < enabledLayers.Count; i++)
+                ppEnabledLayers[i] = Marshal.StringToHGlobalAnsi(enabledLayers[i]);
+
+            var instanceCreateInfo = new InstanceCreateInfo
+            {
+                SType = StructureType.InstanceCreateInfo,
+                PApplicationInfo = &applicationInfo,
+                PpEnabledExtensionNames = (byte**)ppEnabledExtensions,
+                PpEnabledLayerNames = (byte**)ppEnabledLayers,
+                EnabledExtensionCount = (uint)enabledExtensions.Count,
+                EnabledLayerCount = (uint)enabledLayers.Count
+            };
+
+            api.CreateInstance(in instanceCreateInfo, null, out var instance).ThrowOnError();
+            
+            Marshal.FreeHGlobal(applicationName);
+            Marshal.FreeHGlobal(engineName);
+
+            for (var i = 0; i < enabledExtensions.Count; i++) Marshal.FreeHGlobal(ppEnabledExtensions[i]);
+
+            for (var i = 0; i < enabledLayers.Count; i++) Marshal.FreeHGlobal(ppEnabledLayers[i]);
+
+            if (options.UseDebug && api.TryGetInstanceExtension(instance, out ExtDebugUtils debugUtils))
+            {
+                var createInfo = new DebugUtilsMessengerCreateInfoEXT
+                {
+                    SType = StructureType.DebugUtilsMessengerCreateInfoExt,
+                    MessageSeverity = DebugUtilsMessageSeverityFlagsEXT.DebugUtilsMessageSeverityVerboseBitExt | DebugUtilsMessageSeverityFlagsEXT.DebugUtilsMessageSeverityWarningBitExt | DebugUtilsMessageSeverityFlagsEXT.DebugUtilsMessageSeverityErrorBitExt,
+                    MessageType = DebugUtilsMessageTypeFlagsEXT.DebugUtilsMessageTypeGeneralBitExt | DebugUtilsMessageTypeFlagsEXT.DebugUtilsMessageTypeValidationBitExt | DebugUtilsMessageTypeFlagsEXT.DebugUtilsMessageTypePerformanceBitExt,
+                    PfnUserCallback = new PfnDebugUtilsMessengerCallbackEXT(LogCallback),
+                };
+
+                debugUtils.CreateDebugUtilsMessenger(instance, createInfo, null, out var messenger);
+            }
+
+            return new VulkanInstance(instance, api);
+        }
+
+        private static unsafe uint LogCallback(DebugUtilsMessageSeverityFlagsEXT messageSeverity, DebugUtilsMessageTypeFlagsEXT messageTypes, DebugUtilsMessengerCallbackDataEXT* pCallbackData, void* pUserData)
+        {
+            if (messageSeverity >= DebugUtilsMessageSeverityFlagsEXT.DebugUtilsMessageSeverityWarningBitExt)
+            {
+                var message = Marshal.PtrToStringAnsi((nint)pCallbackData->PMessage);
+
+                Logger.TryGet(LogEventLevel.Warning, "Vulkan")?.Log(null, message);
```

Comment:
Should add a mapping from Vulkan message severity to Avalonia. A simple switch will do.